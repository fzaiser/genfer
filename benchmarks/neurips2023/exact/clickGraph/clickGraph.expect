Support is a subset of: [0, 1]

Computing moments...
Total measure:             Z = 0.0022552029749657062
Expected value:            E = 0.6141541855827569
Standard deviation:        Ïƒ = 0.27310054838653336
Variance:                  V = 0.07458390952902524
Skewness (3rd std moment): S = -0.4825243134655868
Kurtosis (4th std moment): K = 2.1400837563718556
