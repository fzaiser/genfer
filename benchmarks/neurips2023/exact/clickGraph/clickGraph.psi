// flags: --expectation
def main() {
    similarityAll := uniform(0, 1);
    beta1 := 0;
    beta2 := 0;

    // i = 0
    if 1 == bernoulli(similarityAll) {
        beta1 = uniform(0, 1);
        beta2 = beta1;
    } else {
        beta1 = uniform(0, 1);
        beta2 = uniform(0, 1);
    }
    observe(1 == bernoulli(beta1));
    observe(1 == bernoulli(beta2));

    // i = 1
    if 1 == bernoulli(similarityAll) {
        beta1 = uniform(0, 1);
        beta2 = beta1;
    } else {
        beta1 = uniform(0, 1);
        beta2 = uniform(0, 1);
    }
    observe(1 == bernoulli(beta1));
    observe(1 == bernoulli(beta2));

    // i = 2
    if 1 == bernoulli(similarityAll) {
        beta1 = uniform(0, 1);
        beta2 = beta1;
    } else {
        beta1 = uniform(0, 1);
        beta2 = uniform(0, 1);
    }
    observe(1 == bernoulli(beta1));
    observe(1 == bernoulli(beta2));

    // i = 3
    if 1 == bernoulli(similarityAll) {
        beta1 = uniform(0, 1);
        beta2 = beta1;
    } else {
        beta1 = uniform(0, 1);
        beta2 = uniform(0, 1);
    }
    observe(0 == bernoulli(beta1));
    observe(0 == bernoulli(beta2));

    // i = 4
    if 1 == bernoulli(similarityAll) {
        beta1 = uniform(0, 1);
        beta2 = beta1;
    } else {
        beta1 = uniform(0, 1);
        beta2 = uniform(0, 1);
    }
    observe(0 == bernoulli(beta1));
    observe(0 == bernoulli(beta2));

    return similarityAll;
}