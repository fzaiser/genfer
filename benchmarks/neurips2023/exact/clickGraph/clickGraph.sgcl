similarityAll ~ UniformCont(0, 1);

# i = 0
if 1 ~ Bernoulli(similarityAll) {
    beta1 ~ UniformCont(0, 1);
    beta2 := beta1;
} else {
    beta1 ~ UniformCont(0, 1);
    beta2 ~ UniformCont(0, 1);
}
observe 1 ~ Bernoulli(beta1);
observe 1 ~ Bernoulli(beta2);

# i = 1
if 1 ~ Bernoulli(similarityAll) {
    beta1 ~ UniformCont(0, 1);
    beta2 := beta1;
} else {
    beta1 ~ UniformCont(0, 1);
    beta2 ~ UniformCont(0, 1);
}
observe 1 ~ Bernoulli(beta1);
observe 1 ~ Bernoulli(beta2);

# i = 2
if 1 ~ Bernoulli(similarityAll) {
    beta1 ~ UniformCont(0, 1);
    beta2 := beta1;
} else {
    beta1 ~ UniformCont(0, 1);
    beta2 ~ UniformCont(0, 1);
}
observe 1 ~ Bernoulli(beta1);
observe 1 ~ Bernoulli(beta2);

# i = 3
if 1 ~ Bernoulli(similarityAll) {
    beta1 ~ UniformCont(0, 1);
    beta2 := beta1;
} else {
    beta1 ~ UniformCont(0, 1);
    beta2 ~ UniformCont(0, 1);
}
observe 0 ~ Bernoulli(beta1);
observe 0 ~ Bernoulli(beta2);

# i = 4
if 1 ~ Bernoulli(similarityAll) {
    beta1 ~ UniformCont(0, 1);
    beta2 := beta1;
} else {
    beta1 ~ UniformCont(0, 1);
    beta2 ~ UniformCont(0, 1);
}
observe 0 ~ Bernoulli(beta1);
observe 0 ~ Bernoulli(beta2);

return similarityAll;
