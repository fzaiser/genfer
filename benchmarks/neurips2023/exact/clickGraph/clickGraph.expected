E = 0.614154185
14475/23569